src/Material.h

```diff
@@ -190,9 +231,72 @@ class Material {
  * @param origin the column index of the matrix element
  * @param destination the row index of the matrix element
  */
-inline FP_PRECISION Material::getSigmaSByGroupInline(
+inline FP_PRECISION MacroMaterial::getSigmaSByGroupInline(
           int origin, int destination) {
   return _sigma_s[destination*_num_groups + origin];
 }
 
+
+/**
+ * @brief inline function for efficient mapping for scattering, from
+ *        1D as stored in memory to 2D matrix
+ * @details Encapsulates the logic for indexing into the scattering
+ *        matrix so it does not need to be repeated in other parts of 
+ *        the code.  Note that this routine is 0-based, rather than 
+ *        1-based indexing, as it is intended for use inside the code,
+ *        not by users from Python.
+ * @param xs the cross section value
+ * @param origin the column index of the matrix element
+ * @param destination the row index of the matrix element
+ */
+inline void MacroMaterial::setSigmaSByGroupInline(
+          double xs, int origin, int destination) {
+  _sigma_s[destination*_num_groups + origin] = xs;
+}
+
+
+
+/**
+ * @class IsoMaterial Material.h "src/Material.h"
+ * @brief The IsoMaterial class represents a unique material, defined as a set
+ *        of Isotopes.
+ */
+class IsoMaterial : public Material {
+
+private:
+  /** The vector of Isotopes contained in the Material */
+  std::vector<Isotope*> _isotopes;
+  
+  /** The vector of number densities associated with the Isotopes */
+  std::vector<FP_PRECISION> _num_dens;
+  
+  /** The number of Isotopes in the Material */
+  int _num_isotopes;
+  
+  /** A flag for whether the fission spectrum has been set */
+  bool _chi_set;
```

Comment:
Why would you move chi from the ``Isotopes`` to the ``IsoMaterial``? It seems logical to me to keep it isotopic. And related to one of my other comments, perhaps instead of a ``Material:getFissionSource()`` routine you could use a ``Material::getChiFissionSource()`` routine. In the case of ``MacroMaterials``, this would use the macroscopic chi, or in the case of ``IsoMaterials``, it would use each ``Isotopes`` chi separately. This would have the effect of abstracting chi from the ``Solver's`` and making its routines simpler and more generalizable for different material formulations.